twython/endpoints.py

```diff
@@ -199,7 +199,28 @@ def upload_video(self, media, media_type, size=None):
             'command': 'FINALIZE',
             'media_id': media_id
         }
-        return self.post(upload_url, params=params)
+        response_finalize = self.post(upload_url, params=params)
+
+        # Stage 4: STATUS call if still processing
+        params = {
+            'command': 'STATUS',
+            'media_id': media_id
+        }
+        
+        processing_state = response_finalize['processing_info'].get('state', None)
+
+        if processing_state is not None:
+            while (processing_state == "pending" or processing_state == "in_progress") :
+                # get the secs to wait
+                check_after_secs = response_finalize['processing_info'].get('check_after_secs', None)
```

Comment:
You don't need to default to `None` as `check_after_secs = response_finalize['processing_info'].get('check_after_secs')` will return `None` if it doesn't exist